src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java

```diff
@@ -22,10 +23,14 @@ public JsonConfigCollection parseFiles(File baseDir, String[] files) {
             InputStream inputStream = null;
             try {
                 inputStream = new FileInputStream(new File(baseDir, file));
-                YamlReader reader = new YamlReader(new InputStreamReader(inputStream));
+                YamlConfig config = new YamlConfig();
+                config.setAllowDuplicates(false);
+                YamlReader reader = new YamlReader(new InputStreamReader(inputStream), config);
                 Object rootObject = reader.read();
                 JsonConfigCollection filePart = rootTransform.transform(rootObject, file);
                 collection.append(filePart);
+            } catch (YamlReader.YamlReaderException ex) {
+                collection.addError("YAML contains duplicate keys: " + ex.getMessage() , file);
```

Comment:
you are right!! how about just directly adding the YamlReaderException (or better YamlException) message, whatever it is and surface it to the user. This would also help users fix if there are other genuine issues with their YAML caught by YamlBeans? e.g below:

```java
} catch (YamlReader.YamlReaderException ex) {
   collection.addError(ex.getMessage() , file);
}
````

The assertion in test that I added in that case becomes:

````java
assertThat(errors.get(0).getAsJsonObject().getAsJsonPrimitive("message").getAsString(), is("Line 9, column 20: Duplicate key found 'upstream'"));
// "Line 9, column 20: Duplicate key found 'upstream'" is directly thrown by YamlBeans library now
````